tests/test_csc.py

```diff
@@ -21,35 +21,39 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import pathlib
 import unittest
 
 from lsst.ts import salobj
-
 from lsst.ts.weatherstation import csc
 
-index_gen = salobj.index_generator()
+TEST_CONFIG_DIR = pathlib.Path(__file__).parents[1].joinpath("tests", "data", "config")
 
 
 class CscTestCase(salobj.BaseCscTestCase, unittest.IsolatedAsyncioTestCase):
     def basic_make_csc(self, initial_state, config_dir, simulation_mode, **kwargs):
         return csc.CSC(
-            index=next(index_gen),
+            index=1,
```

Comment:
I gently suggest providing index as an argument to basic_make_csc. It does, I admit, require specifying index in all calls to make_csc (unless you override make_csc itself to provide useful defaults, which is a bit of a hassle), but I think it is worth considering for two reasons:

* It allows you to test other indices (if you add support for them to data/config/_init.yaml)
* At present you are silently ignoring the existing index argument to make_csc, which could be confusing for future code maintainers.

Reply:
Done